test/extended/util/framework.go

```diff
@@ -267,6 +322,80 @@ func CreatePodForImage(dockerImageReference string) *kapi.Pod {
 	}
 }
 
+// CreatePersistentVolume creates a HostPath Persistent Volume.
+func CreatePersistentVolume(name, capacity, hostPath string) *kapi.PersistentVolume {
+	return &kapi.PersistentVolume{
+		TypeMeta: kapi.TypeMeta{
+			Kind:       "PersistentVolume",
+			APIVersion: "v1",
+		},
+		ObjectMeta: kapi.ObjectMeta{
+			Name:   name,
+			Labels: map[string]string{"name": name},
+		},
+		Spec: kapi.PersistentVolumeSpec{
+			PersistentVolumeSource: kapi.PersistentVolumeSource{
+				HostPath: &kapi.HostPathVolumeSource{
+					Path: hostPath,
+				},
+			},
+			Capacity: kapi.ResourceList{
+				kapi.ResourceStorage: resource.MustParse(capacity),
+			},
+			AccessModes: []kapi.PersistentVolumeAccessMode{
+				kapi.ReadWriteOnce,
+				kapi.ReadOnlyMany,
+				kapi.ReadWriteMany,
+			},
+		},
+	}
+}
+
+// SetupHostPathVolumes will create multiple PersistentVolumes with given capacity
+func SetupHostPathVolumes(c kclient.PersistentVolumeInterface, prefix, capacity string, count int) (volumes []*kapi.PersistentVolume, err error) {
+	rootDir, err := ioutil.TempDir(TestContext.OutputDir, "persistent-volumes")
+	if err != nil {
+		return volumes, err
+	}
+	for i := 0; i < count; i++ {
+		dir, err := ioutil.TempDir(rootDir, fmt.Sprintf("%0.4d", i))
+		if err != nil {
+			return volumes, err
+		}
+		if _, err = exec.LookPath("chcon"); err != nil {
+			err := exec.Command("chcon", "-t", "svirt_sandbox_file_t", dir).Run()
+			if err != nil {
+				return volumes, err
+			}
+		}
+		if err = os.Chmod(dir, 0777); err != nil {
+			return volumes, err
+		}
+		pv, err := c.Create(CreatePersistentVolume(fmt.Sprintf("%s%s-%0.4d", pvPrefix, prefix, i), capacity, dir))
+		if err != nil {
+			return volumes, err
+		}
+		volumes = append(volumes, pv)
+	}
+	return volumes, err
+}
+
+// CleanupHostPathVolumes removes all PersistentVolumes created by
+// SetupHostPathVolumes, with a given prefix
+func CleanupHostPathVolumes(c kclient.PersistentVolumeInterface, prefix string) error {
+	pvs, err := c.List(labels.Everything(), nil)
+	if err != nil {
+		return err
+	}
+	prefix = fmt.Sprintf("%s%s-", pvPrefix, prefix)
+	for _, pv := range pvs.Items {
+		if strings.HasPrefix(pv.Name, prefix) {
+			c.Delete(pv.Name)
```

Comment:
Not sure if this would be helpful here as I'm not familiar with the context, but it came to mind that we can have OpenShift cleanup PVs for us:
https://docs.openshift.org/latest/admin_guide/persistent_storage_nfs.html#provisioning

```json
 "persistentVolumeReclaimPolicy": "Recycle"
```